fuel-android/src/test/kotlin/com/github/kittinunf/fuel/android/RequestAndroidAsyncTest.kt

```diff
@@ -267,8 +332,8 @@ class RequestAndroidAsyncTest : BaseTestCase() {
         assertThat(res, notNullValue())
         assertThat(err, nullValue())
         assertThat(data, notNullValue())
-        assertThat(data as HttpBinHeadersModel, isA(HttpBinHeadersModel::class.java))
-        assertThat((data as HttpBinHeadersModel).headers.isNotEmpty(), isEqualTo(true))
+        assertThat(data!!.headers.isNotEmpty(), isEqualTo(true))
```

Comment:
if it worth putting these tests in 

``` kotlin
if (data != null){

}```

so we can remove the !!

Reply:
Line 334 will fail if it's null. We basically should add "assert data != null" at line 335 so the kotlin compiler knows that the code would have stopped if it's... null.

Wrapping just increases complexity whilst not solving anything because of line 334.

Reply:
I guess you are asserting not null in the line above (334)

Reply:
Exactly, so the compiler doesn't understand the test assertion, but we know it will ALWAYS be not null if it passes the line above.